### PR TITLE
Master

### PR DESCRIPTION
**Pull request by LepiKlark**
Adding unit test project with a couple of unit tests examples.

### CODE REVIEW

File: DbgEngTest/NativeDumpTest.cs [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+        [TestMethod]
+        public void TestThreadCount()
+        {
+            Assert.AreEqual(Thread.All.Length, 1, "Thread count should equal 1");
+        }
+
+        [TestMethod]
+        public void TestStackLength()
+        {
+            Assert.AreEqual(Thread.Current.StackTrace.Frames.Length, 8, "Stack length should equal 1");
```
Comment (southpolenator) [inline DbgEngTest/NativeDumpTest.cs]: equal 8?

File: DbgEngTest/NativeDumpTest.cs [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+        }
+
+        [TestMethod]
+        public void TestCurrentFrame()
+        {
+            Assert.IsTrue(Thread.Current.StackTrace.CurrentFrame.SourceFileName.Contains("nativedumptest.cpp"));
```
Comment (southpolenator) [inline DbgEngTest/NativeDumpTest.cs]: EndsWith?

File: DbgEngTest/NativeDumpTest.cs [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+        }
+
+        [TestMethod]
+        public void TestModuleExtraction()
+        {
+            Assert.AreEqual(Module.All.Length, 11, "Invalid number of open modules.");
```
Comment (southpolenator) [inline DbgEngTest/NativeDumpTest.cs]: This is kind of unstable. Anti virus can add modules, so it is machine dependent... Just leave second assert, for now...

File: DbgEngTest/NativeDumpTest.cs (hunk 1 above)
Comment (LepiKlark) [inline DbgEngTest/NativeDumpTest.cs]: Yes...

File: DbgEngTest/NativeDumpTest.cs (hunk 2 above)
Comment (LepiKlark) [inline DbgEngTest/NativeDumpTest.cs]: Done.

File: DbgEngTest/NativeDumpTest.cs (hunk 3 above)
Comment (LepiKlark) [inline DbgEngTest/NativeDumpTest.cs]: Yes true.

**Pull request merged**